File: easy/922. Sort Array By Parity II/solution.py
class Solution:
    def sortArrayByParityII(self, nums: List[int]) -> List[int]:
        even = 0
        odd = 1
        
        while even < len(nums) and odd < len(nums):
            if nums[even] % 2 == 0:
                even += 2
            else:
                nums[even], nums[odd] = nums[odd], nums[even]
                odd += 2
        
        return nums

File: easy/922. Sort Array By Parity II/solution.cpp
class Solution {
public:
    vector<int> sortArrayByParityII(vector<int>& nums) {
        int even = 0;
        int odd = 1;
        
        while (even < nums.size() && odd < nums.size()) {
            if (nums[even] % 2 == 0) {
                even += 2;
            } else {
                swap(nums[even], nums[odd]);
                odd += 2;
            }
        }
        
        return nums;
    }
};